src/devices_models/devices/renewable_generation.jl

```diff
@@ -84,151 +86,68 @@ function reactivepower_constraints!(
     return
 end
 
-function activepower_constraints!(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{R},
-    model::DeviceModel{R, <:AbstractRenewableDispatchFormulation},
-    system_formulation::Type{<:PM.AbstractPowerModel},
+function make_active_power_constraints_inputs(
+    ::Type{<:PSY.RenewableGen},
+    ::Type{<:AbstractRenewableDispatchFormulation},
+    ::Type{<:PM.AbstractPowerModel},
     feedforward::Union{Nothing, AbstractAffectFeedForward},
-) where {R <: PSY.RenewableGen}
-    parameters = model_has_parameters(psi_container)
-    use_forecast_data = model_uses_forecasts(psi_container)
-
-    if !parameters && !use_forecast_data
-        constraint_data = Vector{DeviceRange}(undef, length(devices))
-        for (ix, d) in enumerate(devices)
-            name = PSY.get_name(d)
-            ub = PSY.get_activepower(d)
-            limits = (min = 0.0, max = ub)
-            range_data = DeviceRange(name, limits)
-            add_device_services!(range_data, d, model)
-            constraint_data[ix] = range_data
-        end
-        device_range(
-            psi_container,
-            constraint_data,
-            constraint_name(ACTIVE_RANGE, R),
-            variable_name(ACTIVE_POWER, R),
+    use_parameters::Bool,
+    use_forecasts::Bool,
+)
+    if (!use_parameters && !use_forecasts)
+        return DeviceRangeConstraintInputs(;
+            range_constraint_inputs = [RangeConstraintInputs(;
+                constraint_name = ACTIVE_RANGE,
+                variable_name = ACTIVE_POWER,
+                limits_func = x -> (min = 0.0, max = PSY.get_activepower(x)),
```

Comment:
I think this is wrong

Reply:
This is ok, when no parameters and forecasts are used. The Constraint is based on the current operating point.